### PR TITLE
GUACAMOLE-1479: Change the style of disabled users/groups.

### PR DESCRIPTION
**Pull request by necouchman**
This change implements a style change for users and groups that are disabled, changing the font color to gray and italicizing the user or group name.

### CODE REVIEW

File: guacamole/src/main/frontend/src/app/settings/templates/settingsUserGroups.html [hunk 1]
```diff
@@ -34,7 +34,7 @@
                 <td class="user-group-name">
                     <a ng-href="#/manage/{{manageableUserGroup.dataSource | escape}}/userGroups/{{manageableUserGroup.userGroup.identifier | escape}}">
                         <div class="icon user-group"></div>
-                        <span class="name">{{manageableUserGroup.userGroup.identifier}}</span>
+                        <span class="name" ng-class="{'user-group-disabled' : (manageableUserGroup.userGroup.attributes['disabled'] !== null && manageableUserGroup.userGroup.attributes['disabled'] === 'true')}">{{manageableUserGroup.userGroup.identifier}}</span>
```
Comment (mike-jumper) [inline guacamole/src/main/frontend/src/app/settings/templates/settingsUserGroups.html:37]: This level of logic should be abstracted away within the controller.

File: guacamole/src/main/frontend/src/app/settings/templates/settingsUsers.html [hunk 2]
```diff
@@ -43,7 +43,7 @@
                 <td class="username">
                     <a ng-href="#/manage/{{manageableUser.dataSource | escape}}/users/{{manageableUser.user.username | escape}}">
                         <div class="icon user"></div>
-                        <span class="name">{{manageableUser.user.username}}</span>
+                        <span class="name" ng-class="{'user-disabled' : (manageableUser.user.attributes['disabled'] !== null && manageableUser.user.attributes['disabled'] === 'true')}">{{manageableUser.user.username}}</span>
```
Comment (mike-jumper) [inline guacamole/src/main/frontend/src/app/settings/templates/settingsUsers.html:46]: Same here - the logic behind this test should be encapsulated within the controller.

**Review by mike-jumper — CHANGES_REQUESTED** [review]
Looks good, however the `disabled` attribute is currently a custom attribute applied only by the JDBC extensions, so we can't add specific handling for that attribute directly within the webapp as-is.

We could:

* Standardize the attribute at the API level, allowing the webapp to make decisions regarding that attribute.
* Provide a standard API for enabling/disabling a user (ie: an `isDisabled()` and `setDisabled()`) which could then be exposed within the REST API and handled within the UI appropriately.
* Move the handling of styling to the extension defining the attribute.

Given how common enabling/disabling a user or group is likely to be, regardless of backend implementation, I'd lean towards some kind of standardization.

**Comment by necouchman** [comment]
@mike-jumper Okay, I've re-worked things to include moving the "disabled" status to the higher level interfaces and pull it out of the Attributes section. And I think the issues with the controller logic for the CSS style should be more reasonable, now.

**Comment by necouchman** [comment]
@mike-jumper Gentle nudge...

**Comment by necouchman** [comment]
@mike-jumper Ping...

File: guacamole-ext/src/main/java/org/apache/guacamole/net/auth/User.java [hunk 3]
```diff
@@ -79,6 +79,23 @@ public static class Attribute {
...
+    public boolean isDisabled();
+    
+    /**
+     * Set the disabled status of this account to the boolean parameter as
+     * provided, true if the account should be disabled, otherwise false.
+     * 
+     * @param disabled 
+     *     True if the account should be disabled, otherwise false.
+     */
+    public void setDisabled(boolean disabled);
```
[the remark this answers is not in the document]
Comment (necouchman) [reply guacamole-ext/src/main/java/org/apache/guacamole/net/auth/User.java]: Implemented concurrent with move to `Identifiable` interface.

[the remark this answers is not in the document]
Comment (necouchman) [reply guacamole-ext/src/main/java/org/apache/guacamole/net/auth/User.java]: Yeah, makes sense - I moved them to the `Identifiable` interface and implemented the default methods.

File: guacamole-ext/src/main/java/org/apache/guacamole/net/auth/Identifiable.java [hunk 4]
```diff
@@ -43,5 +43,26 @@ public interface Identifiable {
...
+    }
+    
+    /**
+     * Set the disabled status of this object to the boolean value provided,
+     * true if the object should be disabled, otherwise false.
+     * 
+     * @param disabled 
+     *     True if the object should be disabled, otherwise false.
+     */
+    default public void setDisabled(boolean disabled) {
```
[the remark this answers is not in the document]
Comment (necouchman) [reply guacamole-ext/src/main/java/org/apache/guacamole/net/auth/Identifiable.java:64]: Okay - I'm fine creating a new interface for this. Is `Disableable` what we want to go with, or are there other things besides just the ability to disable something that would fit in the new Interface? I can't think of anything necessarily, just wondering before we make it official...

[the remark this answers is not in the document]
Comment (necouchman) [reply guacamole-ext/src/main/java/org/apache/guacamole/net/auth/Identifiable.java]: Went ahead and implemented as `Disableable`, but if anyone thinks of something else it should be shifted to, it should be pretty easy to rename...

**Comment by necouchman** [comment]
@mike-jumper Ping...

**Review by mike-jumper — APPROVED** [review]

**Pull request merged**